src/core/simulator/visualization/default_visualization.cpp

```diff
@@ -30,7 +30,7 @@ namespace argos {
          /* Use real-time clock and set time structures */
          m_tStepFunction = &CDefaultVisualization::RealTimeStep;
          timerclear(&m_tStepClockTime);
-         m_tStepClockTime.tv_usec = 1e6 * CPhysicsEngine::GetSimulationClockTick();
+         m_tStepClockTime.tv_usec = static_cast<UInt32>(1e6 * CPhysicsEngine::GetSimulationClockTick());
```

Comment:
Are these casts really necessary? Is it a warning that makes them necessary?

Reply:
Yes--you get a float -> int conversion warning without the cast.